WcaOnRails/app/webpacker/javascript/hooks/useSaveData.js

```diff
@@ -0,0 +1,35 @@
+import { useState, useCallback } from 'react';
+import { fetchJsonOrError } from '../requests/fetchWithAuthenticityToken';
+
+const throwError = (err) => { throw err; };
+
+// This is a hook that can be used to save some data to the website (as json)
+// It assumes that 'url' is a valid, PATCH-able, url.
+// Example of usage:
+// const { save, saving } = useSaveData();
+// // or if you want to override some options:
+// const { save, saving } = useSaveData({ method: 'DELETE' });
+// // and then:
+// save(modelUrl(), { /* model attrs */ }, () => console.log("success"));
+const useSaveData = (options = {}) => {
+  const [saving, setSaving] = useState(false);
+
+  const save = useCallback((url, data, onSuccess, onError = throwError) => {
+    setSaving(true);
+    fetchJsonOrError(url, {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'PATCH',
+      body: JSON.stringify(data),
+      ...options,
+    }).then(onSuccess).catch(onError).finally(() => setSaving(false));
```

Comment:
More general remark about error handling: Why are we passing a generic callback in `useSaveData` but not in `useLoadedData`? The latter only has one high-level "check whether there was an error at all" implementation and any additional error handling has to be added on top of that.

Reply:
I guess I didn't think about passing a custom callback for `useLoadedData` when creating it; technically I guess it would be more accurate to change `useLoadedData`, as it provide more flexibility to let the user specify the behavior they want.

Reply:
Okay, so we have to consider changing `useLoadedData` after this is changed.
Resolving the commit because that change should not happen as part of this PR.